lib/class/api.class.php

```diff
@@ -792,18 +1185,33 @@ public static function localplay($input)
         $localplay = new Localplay(AmpConfig::get('localplay_controller'));
         $localplay->connect();
 
+        //Whatever format the user wants
+        $outputFormat = $input['format'];
+
+
         switch ($input['command']) {
             case 'next':
             case 'prev':
             case 'play':
             case 'stop':
                 $result_status = $localplay->$input['command']();
-                $xml_array     = array('localplay'=>array('command'=>array($input['command']=>make_bool($result_status))));
-                echo XML_Data::keyed_array($xml_array);
+                $outputArray     = array('localplay'=>array('command'=>array($input['command']=>make_bool($result_status))));
+                if ($outputFormat == 'json') {
+                    echo json_encode($outputArray, JSON_PRETTY_PRINT);
```

Comment:
Just noticed the mix of `json_encode` calls and `JSON_Data` methods calls. This is a bit confusing. I think the methods from `XML_Data` should be replicated in `JSON_Data` and we should call the same methods either from XML or JSON. Of course, most of the time, the `JSON_Data` method will simply be a wrapper around `json_encode`.